packages/vite/src/hbs.ts

```diff
@@ -130,8 +132,10 @@ async function maybeSynthesizeComponentJS(context: PluginContext, source: string
   };
 }
 
+const hbsFilter = createFilter('**/*.hbs?(\\?)*');
+
 function maybeRewriteHBS(resolution: ResolvedId) {
-  if (!resolution.id.split('?')[0].endsWith('.hbs')) {
+  if (!hbsFilter(resolution.id)) {
```

Comment:
@mansona this will match  `/rewritten-app/components/old/template.hbs?v=45cfc9c0/..%2Fcomponent.js%3Fv%3D45cfc9c0-embroider-pair-component`

Reply:
Doesn't your test cover this issue? 🤔

Reply:
mm, its passing, ill have a look